backend/dal/Services/LeaseService.cs

```diff
@@ -1,294 +1,20 @@
-            Context.Entry(existingLease).CurrentValues.SetValues(lease);
-            if(commitTransaction)
-            {
-                this.Context.CommitTransaction();
-            }
-            return existingLease;
-        }
-
-        /// <summary>
-        /// update the tenants on the lease
-        /// </summary>
-        /// <param name="leaseId"></param>
-        /// <param name="pimsLeaseTenants"></param>
-        /// <returns></returns>
-        public PimsLease UpdateLeaseTenants(long leaseId, long rowVersion, ICollection<PimsLeaseTenant> pimsLeaseTenants)
-        {
-            this.User.ThrowIfNotAuthorized(Permissions.LeaseEdit);
-            var existingLease = this.Context.PimsLeases.Include(l => l.PimsLeaseTenants).Where(l => l.LeaseId == leaseId).AsNoTracking().FirstOrDefault()
-                 ?? throw new KeyNotFoundException();
-            if(existingLease.ConcurrencyControlNumber != rowVersion) throw new DbUpdateConcurrencyException("Unable to save. Please refresh your page and try again");
-
-            this.Context.UpdateChild<PimsLease, long, PimsLeaseTenant>(l => l.PimsLeaseTenants, leaseId, pimsLeaseTenants.ToArray());
-            this.Context.CommitTransaction();
-
-            return Get(existingLease.LeaseId);
-        }
-
-        /// <summary>
-        /// update the tenants on the lease
-        /// </summary>
-        /// <param name="leaseId"></param>
-        /// <param name="pimsPropertyImprovements"></param>
-        /// <returns></returns>
-        public PimsLease UpdateLeaseImprovements(long leaseId, long rowVersion, ICollection<PimsPropertyImprovement> pimsPropertyImprovements)
+        IPimsRepository _pimsRepository;
+        public LeaseService(IPimsRepository pimsRepository)
         {
-            this.User.ThrowIfNotAuthorized(Permissions.LeaseEdit);
-            var existingLease = this.Context.PimsLeases.Include(l => l.PimsPropertyImprovements).Where(l => l.LeaseId == leaseId).AsNoTracking().FirstOrDefault()
-                 ?? throw new KeyNotFoundException();
-            if (existingLease.ConcurrencyControlNumber != rowVersion) throw new DbUpdateConcurrencyException("Unable to save. Please refresh your page and try again");
-
-            this.Context.UpdateChild<PimsLease, long, PimsPropertyImprovement>(l => l.PimsPropertyImprovements, leaseId, pimsPropertyImprovements.ToArray());
-            this.Context.CommitTransaction();
-
-            return Get(existingLease.LeaseId);
+            _pimsRepository = pimsRepository;
         }
-
-        /// <summary>
-        /// update the properties on the lease
-        /// </summary>
-        /// <param name="leaseId"></param>
-        /// <param name="pimsPropertyLeases"></param>
-        /// <returns></returns>
-        public PimsLease UpdatePropertyLeases(long leaseId, long rowVersion, ICollection<PimsPropertyLease> pimsPropertyLeases, bool userOverride = false)
+        public void CheckRowVersion(long leaseId, long rowVersion)
```

Comment:
instead of throwing, I think this version should return a boolean. What to do about the values not matching sounds like a higher-level decision than this method